src/solve.jl

```diff
@@ -717,13 +731,26 @@ for `results(R; onlynonsingular=true, conditions...)`. For the possible `conditi
 nonsingular(R::Results; kwargs...) = results(R; onlynonsingular=true, kwargs...)
 
 """
-    singular(result::Results; conditions...)
+    singular(R::Results; singulartol=1e10, multiplicitytol=1e-5, multiple_results=false, kwargs...)
 
-Return all `PathResult`s for which the solution is singular. This is just a shorthand
-for `results(R; onlysingular=true, conditions...)`. For the possible `conditions` see [`results`](@ref).
+Return all `PathResult`s for which the solution is singular. A solution is labeled singular if the condition number is greater than `singulartol`, or if the winding number is > 1. If `multiple_results=false` only one point from each cluster of multiple points (computed from [`multiplicities`](@ref) with tolerance `multiplicitytol`) is returned. If If `multiple_results=true` all singular solutions in `R` are returned.  For the possible `kwargs` see [`results`](@ref).
 """
-function singular(R::Results; singulartol=1e10, tol=singulartol, kwargs...)
-    results(R; onlysingular=true, singulartol=tol, kwargs...)
+function singular(R::Results; singulartol=1e10, multiplicitytol=1e-5, multiple_results=false, kwargs...)
+    S = results(R; onlysingular = true, singulartol=singulartol, kwargs...)
+    if isempty(S) || multiple_results
+        return S
+    else
+        info = multiplicities_info(S, multiplicitytol)
+        S₁ = similar(S, 0)
+        for k in keys(info)
+            for v in info[k]
+                P = deepcopy(S[v[1]])
```

Comment:
Do we need the deepcopy here? I think it is fine to just modify the multiplicity entry.

Reply:
I think a new instance should be created here. Think of the following scenario:

multiplicity_tol = x  gives multiplicity a, but 
multiplicity_tol = y  gives multiplicity b!=a. 

Without deepcopy, the multiplicity is overwritten globally. I dont think this is what we want.